core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -497,13 +499,23 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeException(failure);
         }
 
-        @Override
-        public String toString() {
-            List<String> components = new ArrayList<>(4);
+        private void addSummaryComponents(List<String> components) {
```

Comment:
NIT - I tend not to like that level of code reuse. It just makes it hard to see how the end result looks like. I also think we can just use string concat and be done with it.

Reply:
Noted - no problem.